lib/offsite_payments/integrations/bit_pay.rb

```diff
@@ -84,7 +84,10 @@ def create_invoice
           request = Net::HTTP::Post.new(uri.request_uri)
           request.content_type = "application/json"
           request.body = @fields.to_json
-          request.basic_auth @account, ''
+
+          unless v2_api_token?(@account)
```

Comment:
Are you sure about the `unless`? It means that it will only add the basic auth when it's an api v2 token. Also please add some unit tests to cover this case. Thanks.

Reply:
@pi3r in this statement, when v2_api_token? is true (so v2 token) the inner code won't be run

Reply:
Yeah you're right my bad... Can you add tests though?

Reply:
@pi3r the current tests should be enough for this matter, what else do you think we should cover?

Reply:
The current test don't cover this condition branch. You can change this https://github.com/activemerchant/offsite_payments/blob/master/test/unit/integrations/bit_pay/bit_pay_helper_test.rb#L52 to look like:
```ruby
    stub_request(:post, BitPay::API_V1_URL).with(
      body: {
        orderID: '1234',
        price: '500',
        currency: 'USD',
        posData: { orderId: 1234 }.to_json,
        fullNotifications: "true",
        transactionSpeed: 'high',
        token: @token_v1,
      }.to_json,
      basic_auth: [@token_v1, ''],
    ).to_return(
      status: 200,
      body: { id: @invoice_id }.to_json
    )
```

Reply:
Can you add a description to you pr and we'll be able to merge it?

Reply:
@pi3r Done, thanks